panic: time: missing Location in call to Date

hi, 
i got a panic on http://rpayc.com.au/news-events/event-calendar-list?format=raw&type=ical

panic: time: missing Location in call to Date

goroutine 26 [running]:
time.Date(0x7e2, 0x8, 0x16, 0xe, 0x0, 0x0, 0x0, 0x0, 0xc8f1, 0x15, ...)
        /opt/go1.10.2/src/time/time.go:1313 +0x659
time.parse(0x6da4da, 0xf, 0xc420238759, 0xf, 0x0, 0x0, 0x4, 0x0, 0x70ce60, 0xc420096e50, ...)
        /opt/go1.10.2/src/time/format.go:1094 +0x2ab8
time.ParseInLocation(0x6da4da, 0xf, 0xc420238759, 0xf, 0x0, 0x1, 0xed307259e, 0x854460, 0xc420069320, 0xc420069320)
        /opt/go1.10.2/src/time/format.go:783 +0x62
github.com/luxifer/ical.parseDate(0xc4200997d0, 0x854460, 0x6d93a6, 0xb, 0xc4200693b8, 0x0, 0x0)
        /home/andreas/go/src/github.com/luxifer/ical/parse.go:523 +0x10e
github.com/luxifer/ical.(*parser).validateEvent(0xc420249980, 0xc4200e02c0, 0xc4201ce000, 0x4)
        /home/andreas/go/src/github.com/luxifer/ical/parse.go:426 +0x723
github.com/luxifer/ical.(*parser).scanDelimiter(0xc420249980, 0xf, 0x229, 0xc420238829, 0xa, 0xc4200a6480, 0x4)
        /home/andreas/go/src/github.com/luxifer/ical/parse.go:227 +0x76
github.com/luxifer/ical.(*parser).scanContentLine(0xc420249980, 0x0, 0x0)
        /home/andreas/go/src/github.com/luxifer/ical/parse.go:276 +0x9b6
github.com/luxifer/ical.(*parser).parse(0xc420249980, 0x25ce, 0xc4200a6440, 0x25ce)
        /home/andreas/go/src/github.com/luxifer/ical/parse.go:193 +0xd2
github.com/luxifer/ical.Parse(0x70d320, 0xc42011e5c0, 0x0, 0x4, 0xc4201ee800, 0x26d9)
        /home/andreas/go/src/github.com/luxifer/ical/parse.go:85 +0x1f6
main.parse(0xc4201ec000, 0x26db, 0x26db, 0x3e00, 0xc4201ec000, 0x26db, 0x0)
        /home/andreas/t/crawler.go:182 +0x131
main.handle(0x688d, 0xc4200200a0, 0x48)
        /home/andreas/t/crawler.go:99 +0x47a
main.main.func1()
        /home/andreas/t/crawler.go:45 +0x9c
created by main.main
        /home/andreas/t/crawler.go:56 +0x19e
exit status 2

--- a/fixtures/example.ics
+++ b/fixtures/example.ics
@@ -9,8 +9,8 @@ DTSTART:19960918T143000Z
 DTEND:19960920T220000Z
 STATUS:CONFIRMED
 CATEGORIES:CONFERENCE
-SUMMARY:Networld+Interop Conference
-DESCRIPTION:Networld+Interop Conference and Exhibit\nAtlanta World Congress Center\n Atlanta\, Georgia
+SUMMARY;FOO=bar,"baz":Networld+Interop Conference
+DESCRIPTION;ALTREP="cid:part1.0001@example.org":Networld+Interop Conference and Exhibit\nAtlanta World Congress Center\n Atlanta\, Georgia
 END:VEVENT
 BEGIN:VEVENT
 DTSTAMP:19960704T120000Z

--- a/parse.go
+++ b/parse.go
@@ -519,7 +519,13 @@ func parseDate(prop *Property, l *time.Location) (time.Time, error) {
 	}
 
 	if tz, ok := prop.Params["TZID"]; ok {
-		loc, _ := time.LoadLocation(tz.Values[0])
+		loc, err := time.LoadLocation(tz.Values[0])
+
+		// In case we are not able to load TZID location we default to UTC
+		if err != nil {
+			loc = time.UTC
+		}
+
 		return time.ParseInLocation(dateTimeLayoutLocalized, prop.Value, loc)
 	}
 

--- a/parse_test.go
+++ b/parse_test.go
@@ -102,6 +102,22 @@ func Test_parseDate(t *testing.T) {
 			},
 			want: time.Date(1998, time.January, 19, 7, 0, 0, 0, time.Local),
 		},
+		{
+			name: "Datetime with bad timezone",
+			args: args{
+				prop: &Property{
+					Name: "DTSTART",
+					Params: map[string]*Param{
+						"TZID": &Param{
+							Values: []string{"Z"},
+						},
+					},
+					Value: "19980119T020000",
+				},
+				l: time.Local,
+			},
+			want: time.Date(1998, time.January, 19, 2, 0, 0, 0, time.UTC),
+		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {

